### PR TITLE
Scrollable feature

### PR DESCRIPTION
**Pull request by afsardo**
Fixes: #208, #210 

Adds an onScroll handler, whenever the input scrolls the highlighter scrolls as well.

[x] Passes tests
[x] Documentation added 
[ ] New tests to prove the scroll function

Credits to:
- RelocationOnline.com (My company, where we needed this fix)
- Max LaRue (Co-worker, that had a previous version fix)

### CODE REVIEW

File: demo/src/examples/Scrollable.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import { provideExampleValue } from './higher-order'
+
+import defaultStyle from './defaultStyle'
+import defaultMentionStyle from './defaultMentionStyle'
+
+function Scrollable({ value, data, onChange, onAdd }) {
+  let style = merge({}, defaultStyle, {
+    highlighter: {
+      overflow: 'auto',
```
Comment (jfschwarz) [inline demo/src/examples/Scrollable.js]: This will cause two overlapping scrollbars to be shown. Set `overflow: 'hidden'` instead.

Comment (jfschwarz) [reply demo/src/examples/Scrollable.js]: We might set this in the general default styles of the `highlighter` element. Then it should be enough to just set the custom styles for the `input` element to make the componen scrollable.

**Comment by afsardo** [comment]
I am trying to add the test for when the textarea scrolls, to make sure the Highlighter always has the same `scrollLeft`, `scrollTop`, `height` attributes, but having a really hard time testing it.

**Comment by afsardo** [comment]
Let me know if I can help, this was what I was trying, but doesn't work:

`it.only('should be scrollable if it has fixed height.', () => {
    node.setProps({
      value: "Hi @[John Doe](user:johndoe),\n\n\nlet's add\n\n@[John Doe](user:johndoe) to this conversation...",
      style: {
        input: {
          overflow: 'auto',
          height: 70,
        },
      },
    }).render()

    const textarea = node.find('textarea')
    textarea.scrollLeft = 0
    textarea.scrollTop = 120
    textarea.prop('onScroll')

    const Highlighter = node.find('Highlighter')
    expect(Highlighter.scrollLeft).toEqual(0)
    expect(Highlighter.scrollTop).toEqual(120)
    expect(Highlighter.clientHeight).toEqual(50)
  })`

**Comment by jfschwarz** [comment]
I also had some difficulties writing a test for this as a default enzyme `mount` mounts the component somewhere so that it is not possible to get `scollHeight`, `scrollTop`, `offsetHeight` properties of HTML elements. 😕 Setting `attachTo` config option for `mount` fixed this.

Here's the commit, in case you're interested: https://github.com/signavio/react-mentions/pull/230/commits/01aee0ef85a2b53ecbe3cfc0396730e24cb83f76

Thank you again for your contribution.

**Comment by afsardo** [comment]
Nice one ^^ Yes I encountered some people talking about enzyme no really rendering an element, but wasn’t sure how to fix it, thanks for sharing the commit! TIL :)

**Comment by rutpshah** [comment]
@afsardo Is this feature released yet?

**Pull request merged**